### PR TITLE
greedy search remove softmax function

### PR DESCRIPTION
**Pull request by wtmlon**
### Description

删除贪心搜索中冗余的 softmax 计算

### CODE REVIEW

File: paddlenlp/transformers/generation_utils.py [hunk 1]
```diff
@@ -1061,10 +1061,8 @@ def greedy_search(self, input_ids, logits_processors, max_length, pad_token_id,
             next_token_logits = self.adjust_logits_during_generation(next_token_logits)
             next_tokens_scores = logits_processors(input_ids, next_token_logits)
             # greedy
-            probs = F.softmax(next_tokens_scores)
-            probs = paddle.log(probs)
-            next_tokens = paddle.argmax(probs, axis=-1).unsqueeze(-1)
-            next_scores = paddle.index_sample(probs.astype("float32"), next_tokens)
+            next_tokens = paddle.argmax(next_tokens_scores, axis=-1).unsqueeze(-1)
+            next_scores = paddle.index_sample(next_tokens_scores.astype("float32"), next_tokens)
```
Comment (sijunhe) [inline paddlenlp/transformers/generation_utils.py:1065]: next_scores 这里的逻辑变了，以前的scores是softmax + log过的，现在是原始的logits.

Comment (sijunhe) [reply paddlenlp/transformers/generation_utils.py:1065]: 这块看下transformers的代码是怎么做的
